File: 919-projection-area-of-3d-shapes/projection-area-of-3d-shapes.py
class Solution:
    def projectionArea(self, grid: List[List[int]]) -> int:
        top_projection = sum(1 for row in grid for cell in row if cell > 0)
        front_projection = sum(max(row) for row in grid)

        side_projection = sum(max(column) for column in zip(*grid))

        return top_projection + front_projection + side_projection

File: 919-projection-area-of-3d-shapes/projection-area-of-3d-shapes.js
/**
 * @param {number[][]} grid
 * @return {number}
 */
var projectionArea = function(grid) {
    let area = 0;
    const n = grid.length;

    for (let row = 0; row < n; row++) {
        let rowMax = 0;
        let colMax = 0;

        for (let col = 0; col < n; col++) {
            if (grid[row][col] > 0) {
                area++;
            }
            rowMax = Math.max(rowMax, grid[row][col]);

            colMax = Math.max(colMax, grid[col][row]);
        }
        area += rowMax + colMax;
    }
    return area;
};